### PR TITLE
[iOS][Tailored Onboarding] Present Final dialog for Duck.ai flow

### PR DESCRIPTION
**Pull request by alessandroboron**
Task/Issue URL: https://app.asana.com/1/137249556945/project/72649045549333/task/1214080667430308
Tech Design URL:
CC:

### Description

Adds the Duck.ai-tailored onboarding completion dialog and routes the subscription promo, reusing the existing infrastructure made for the Duck.ai onboarding experiment.

### Key Changes:
- New Duck.ai copy for the end-of-journey completion message and subscription promo (`UserText.Onboarding.DuckAICPP.Contextual`).
- `NewTabPageViewController` now branches on `OnboardingFlowProviding.currentOnboardingFlow`. A new `NewTabPageOnboardingDialogEvent` enum splits the previously-overloaded `presentNextDaxDialog()` into two intents: `linearOnboardingCompleted` (post-modal) and `nextDialogRequested` (everything else). The tailored flow skips the regular Dax sequence and only surfaces the subscription promo when `subscriptionPromotionPending` is set.
- On the tailored flow, the NTP launches the omnibar in `.aiChat` mode (both for the completion CTA and the skip branch) so users land in the Duck.ai experience the onboarding emphasised.
- Subscription promo redirect now accepts an `OnboardingSubscriptionPromotionPage?`; the Duck.ai flow passes `.duckAI`, appending `featurePage=duckai` to the purchase URL. Default flow passes `nil` (unchanged behaviour).
- `MainViewController.onboardingCompleted` waits for `OnboardingIntroViewController` dismissal before calling `newTabPageViewController?.onboardingCompleted()` so the editing-state controller can install. It also enables `aiChatSearchInputUserSettings` for the tailored flow (skipping the broader prerequisite to avoid disabling contextual dialogs mid-chain).
- `presentContextualOnboardingDialogIfNeeded()` gates the post-fire completion dialog to the default flow, preventing the tailored flow from presenting twice.
- `RebrandedContextualOnboardingDialogs+EndOfJourney` now uses `OnboardingRichTextMessageRenderer` for the chat-icon placeholder rendering.

### Testing Steps
**Scenario 1 - Duck.ai flow**
1. In `FeatureFlag.swift` set `Config(source: .remoteReleasable(.feature(.iOSBrowserConfig)))` for `.onboardingDuckAIFlow` 
2. In `CustomProductPage+Onboarding` -> `extension AppStoreCustomProductPageEvaluator` add `let url = URL(string: "ddgCPP://duckAI”)` at line 33 to simulate launching the app from Duck.ai CPP.
3. Fresh install the app and complete the linear onboarding (don't skip).
4. Verify the NTP shows the Duck.ai completion dialog with the chat icon inline, address bar opens in AI-chat mode.
5. Tap the completion dialog's CTA → subscription promo appears with the Duck.ai-specific body copy. 
6. Tap **Get DuckDuckGo Pro** → confirm the opened URL contains `featurePage=duckai`.
7. Repeat step 3-5, this time tapping **Skip** during linear onboarding: verify NTP shows no Dax dialog and the omnibar opens directly in AI-chat mode.

**Scenario 2- Default flow**
1. Remove code added at step 1&2 of Scenario 1.
2. Fresh install the app.
3. Run the default onboarding flow end-to-end and confirm the regular Dax sequence, completion dialog, and promo URL (no `featurePage`) are unchanged.

**Scenario 3- Duck.ai experiment**
1. Smoke test experiment it works as expected.

### Impact and Risks
**Medium**: Touches the onboarding completion handoff and NTP dialog routing for both default and Duck.ai flows.

#### What could go wrong?
- **Default flow regression on the new event router.** Mitigation: `presentDefaultFlowDialog` preserves the previous behaviour (`showNextDaxDialogNew` + `beginEditing` on completion); existing tests still pass.
- **Tailored completion dialog presented twice.** The `presentContextualOnboardingDialogIfNeeded()` gate ensures the post-fire path no-ops in the tailored flow; the explicit `onboardingCompleted` callback is the single entry point.
- **`OmniBarEditingStateViewController` failing to install** because `aiChatSearchInputUserSettings` is off. Mitigated by enabling the setting just-in-time in `onboardingCompleted`.
- **Subscription promo `featurePage` query item drift.** New unit tests pin both the present and absent cases.

### Quality Considerations
- Edge case: user skips linear onboarding in the tailored flow → covered by `MockTutorialSettings.hasSkippedOnboarding` branch and unit test.
- No new pixels or privacy surface; subscription URL change only adds a query parameter visible to the purchase web flow.

### Notes to Reviewer
- This PR does not include **UTI** fixes.
- The two long comments in `MainViewController.onboardingCompleted` and `NewTabPageViewController.showDuckAIOnboardingCompletionWithActiveAddressBar` document non-obvious sequencing constraints; please flag if they're unclear.
- `ensureExperimentCompletionDialogPresentationPrerequisites` was renamed to `ensureDuckAiCompletionDialogPresentationPrerequisites` and made internal so the tailored flow can reference it (it does not call it, see the comment for why).
- Pixel updates will happen in a subsequent PR.

—
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

---

> [!NOTE]
> **Medium Risk**
> Changes onboarding completion timing, NTP Dax routing, and subscription purchase URLs for both default and Duck.ai flows; regressions could break dialog chains or default onboarding.
> 
> **Overview**
> Adds the **Duck.ai tailored** post-linear onboarding path on the new tab page: after the intro modal dismisses, users see Duck.ai-specific end-of-journey copy (with inline chat icon via `OnboardingRichTextMessageRenderer`) and the omnibar opens in **`.aiChat`** mode; skip onboarding lands in AI chat without Dax tips.
> 
> `NewTabPageViewController` now routes NTP dialogs through `OnboardingFlowProviding` and a `NewTabPageOnboardingDialogEvent` split (`linearOnboardingCompleted` vs `nextDialogRequested`). The tailored flow **skips** the regular Dax sequence and only chains the **subscription promo** when `subscriptionPromotionPending` is set, with Duck.ai promo copy and purchase URLs that append `featurePage=duckai`.
> 
> `MainViewController` defers NTP handoff until onboarding intro dismissal, enables AI search input settings for tailored completion without calling `disableContextualDaxDialogs` (to preserve promo → EOJ chaining), and gates post-tab-switch completion to the **default** flow so tailored completion is not shown twice.
> 
> Default onboarding behavior is preserved behind the same event router; tests cover tailored branches and subscription URL query parameters.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f5ccfd54476d6931dda93ca1cfd5616cb6b1aca0. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by mallexxx — APPROVED** [review]
LGTM

Issue that possibly worths attention is the burning animation faded out before completion:
https://github.com/user-attachments/assets/1bbaf0d1-8d03-4810-80c2-c9a940a13d2c

File: iOS/DuckDuckGo/NewTabPageViewController.swift [hunk 1]
```diff
@@ -275,34 +282,71 @@ final class NewTabPageViewController: UIHostingController<NewTabPageView>, NewTa
...
+        switch event {
+        case .nextDialogRequested:
+            // Tailored flow never enters the regular Dax sequence. Only the subscription promo can
+            // surface here — chained from the completion dialog's onDismiss via `showNextDaxDialog()`
+            // after `setFinalOnboardingDialogSeen()` flips `subscriptionPromotionPending` true.
+            presentSubscriptionPromotionIfPending()
+        case .linearOnboardingCompleted:
+            // Skip branch does not show Dax dialogs. Land the user in a new tab page with the AI-chat-mode address bar prompted.
+            if tutorialSettings.hasSkippedOnboarding {
+                chromeDelegate?.omniBar.beginEditing(animated: true, forTextEntryMode: .aiChat)
```
[the remark this answers is not in the document]
Comment (alessandroboron) [reply iOS/DuckDuckGo/NewTabPageViewController.swift:341]: `OnboardingIntroViewModel.confirmSkipOnboardingAction` calls `contextualDaxDialogs.disableContextualDaxDialogs()` so this is not an issue

**Comment by alessandroboron** [comment]
> ossibly worths attention is the burning animation faded out before completion

Thanks @mallexxx, I will check with design during ship review and if needed I could resume the onboarding when the animation completes
